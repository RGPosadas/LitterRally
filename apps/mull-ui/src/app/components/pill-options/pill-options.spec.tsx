import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import PillOptions from './pill-options';

const mockOptions: string[] = ['Yes', 'No', 'Maybe'];
const mockOnChange: jest.Mock = jest.fn();
describe('PillOptions', () => {
  it('should render successfully', () => {
    const { baseElement } = render(
      <PillOptions options={mockOptions} onChange={mockOnChange} active={0} />
    );
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const tree = renderer
      .create(<PillOptions options={mockOptions} onChange={mockOnChange} active={0} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have an active option', () => {
    render(<PillOptions options={mockOptions} onChange={mockOnChange} active={0} />);
    const pill = screen.getByTestId('pill-id-0');
    expect(pill.classList.contains('pill-option-active')).toBeTruthy();
  });

  it('should have a different active option', () => {
    const { rerender } = render(
      <PillOptions options={mockOptions} onChange={mockOnChange} active={0} />
    );
    let pill = screen.getByTestId('pill-id-0');
    expect(pill).toHaveTextContent('Yes');
    pill = screen.getByTestId(`pill-id-1`);
    fireEvent.click(pill);
    expect(mockOnChange).toHaveBeenCalled();
    const activePill = mockOnChange.mock.calls[0][0];
    rerender(<PillOptions options={mockOptions} onChange={mockOnChange} active={activePill} />);
    pill = screen.getByTestId(`pill-id-${activePill}`);
    expect(pill.classList.contains('pill-option-active')).toBeTruthy();
  });
});
