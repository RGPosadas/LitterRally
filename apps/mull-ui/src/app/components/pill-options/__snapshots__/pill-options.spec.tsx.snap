// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PillOptions should match snapshot 1`] = `
<div
  className="pill-options-container"
>
  <label
    className="custom-text-input-label"
  >
    Restriction
  </label>
  <div
    className="pill-options"
  >
    <button
      className="pill-option pill-option-active"
      data-testid="pill-id-0"
      onClick={[Function]}
      type="button"
    >
      Yes
    </button>
    <button
      className="pill-option pill-option-inactive"
      data-testid="pill-id-1"
      onClick={[Function]}
      type="button"
    >
      No
    </button>
    <button
      className="pill-option pill-option-inactive"
      data-testid="pill-id-2"
      onClick={[Function]}
      type="button"
    >
      Maybe
    </button>
  </div>
</div>
`;
